skimage/util/_map_array.py

```diff
@@ -20,6 +20,18 @@ def map_array(input_arr, input_vals, output_vals, out=None):
     -------
     out : array, same shape as `input_arr`
         The array of mapped values.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> import skimage as ski
+    >>> ski.util.map_array(
+    ...    input_arr=np.array([[0, 2, 2, 0], [3, 4, 5, 0]]),
+    ...    input_vals=np.array([1, 2, 3, 4, 6]),
+    ...    output_vals=np.array([6, 7, 8, 9, 10]),
+    ... )
+    array([[0, 7, 7, 0],
+           [8, 9, 0, 0]])
```

Comment:
I was initially confused, why the `5` in this example is replaced with `0` as well because it's not part of `input_vals`. Replace missing values with 0 seems to be undocumented behavior as old and new version do this.

However, for the old version I can't figure out, where in the implementation `5` is replaced with `0`. It's bugging me. :sweat_smile:

Reply:
Not sure why but `lut[5]` seems to return `0`. Maybe that's the default value of [`unordered_map`](https://en.cppreference.com/w/cpp/container/unordered_map). That behavior should definitely be documented in the docstring.

Reply:
it comes from `operator[]` specification https://en.cppreference.com/w/cpp/container/unordered_map/operator_at

In general, for unknown values it inserts an object constructed by no argument constructor. And for int, no argument constructor returns 0. 

But for the new implementation it should not overwrite not mapped values in your provided output array.